src/pydvl/value/shapley/montecarlo.py

```diff
@@ -86,123 +216,139 @@ def truncated_montecarlo_shapley(
-    :param n_iterations: Terminate if the total number of permutations exceeds
-        this number.
+    :param done: Check on the results which decides when to stop
+        sampling permutations.
+    :param truncation: callable that decides whether to stop computing
+        marginals for a given permutation.
     :param n_jobs: number of jobs processing permutations. If None, it will be
         set to :func:`available_cpus`.
-    :param config: Object configuring parallel computation, with cluster address,
-        number of cpus, etc.
-    :param progress: Whether to display progress bars for each job.
-    :param coordinator_update_period: in seconds. Check status with the job
-        coordinator every so often.
-    :param worker_update_period: interval in seconds between different updates to
-        and from the coordinator
+    :param config: Object configuring parallel computation, with cluster
+        address, number of cpus, etc.
+    :param coordinator_update_period: in seconds. How often to check the
+        accumulated results from the workers for convergence.
+    :param worker_update_period: interval in seconds between different
+        updates to and from the coordinator
     :return: Object with the data values.
 
     """
+    # Avoid circular imports
+    from .actor import get_shapley_coordinator, get_shapley_worker
+
     if config.backend == "sequential":
         raise NotImplementedError(
             "Truncated MonteCarlo Shapley does not work with "
             "the Sequential parallel backend."
         )
 
     parallel_backend = init_parallel_backend(config)
-
     n_jobs = parallel_backend.effective_n_jobs(n_jobs)
-
     u_id = parallel_backend.put(u)
 
-    coordinator = get_shapley_coordinator(  # type: ignore
-        value_tolerance, n_iterations, progress, config=config
-    )
+    coordinator = get_shapley_coordinator(config=config, done=done)  # type: ignore
+
     workers = [
         get_shapley_worker(  # type: ignore
             u=u_id,
             coordinator=coordinator,
+            truncation=truncation,
             worker_id=worker_id,
-            progress=progress,
             update_period=worker_update_period,
             config=config,
         )
         for worker_id in range(n_jobs)
     ]
-    for worker_id in range(n_jobs):
-        workers[worker_id].run(block=False)
+    for worker in workers:
+        worker.run(block=False)
 
-    while not coordinator.check_done():
+    while not coordinator.check_convergence():
         sleep(coordinator_update_period)
 
-    values, stderr = coordinator.get_results()
+    return coordinator.accumulate()
 
-    return ValuationResult(
-        algorithm="truncated_montecarlo_shapley",
-        status=coordinator.status(),
-        values=values,
-        stderr=stderr,
-        data_names=u.data.data_names,
-    )
+    # Something like this would be nicer, but it doesn't seem to be possible
```

Comment:
I'll remove this comment, but I don't want to run the pipelines again just yet ... 😅